### DISCUSSION TITLE
Is there any need to add a tifffile library as a dedicated tifreader?

### DISCUSSION THREAD

**Discussion by JavaZeroo** [Project-MONAI/MONAI#6423] [Ideas]
I have tested that reading a large file with PIL takes about 4 times longer than tifffile. 🤯

If possible, I can submit a PR for tif_reader based on tifffile. 🤩

In addition, I wonder why WSIReader was removed, it looks like he has been implemented to read tif files using tifffile. 🤔

>  [Speed Test](https://www.kaggle.com/code/brettolsen/image-loading-speed-test)

**Comment by KumoLiu** [comment]
Hi @JavaZeroo, thanks for your interest here.
We didn't remove `WSIReader`, just put it under monai/data folder, and it still support tifffile backend.
https://github.com/Project-MONAI/MONAI/blob/9c9777751ab4f96e059a6597b9aa7ac6e7ca3b92/monai/data/wsi_reader.py#L475

Thanks!

**Reply by JavaZeroo** [reply]
Thanks! Thats really help a lot! ❤️

**Comment by JavaZeroo** [comment]
How to open single channel tif image? Im using TiffFileWSIReader
```python
transforms.LoadImaged(keys=["image"], reader="TiffFileWSIReader", mode='r+' ),
``` 
I tried `mode='r+'` to avoid channel check, but it is doesn't work. :(

https://github.com/Project-MONAI/MONAI/blob/9c9777751ab4f96e059a6597b9aa7ac6e7ca3b92/monai/data/wsi_reader.py#LL1179C1-L1186C30

**Reply by KumoLiu** [reply]
Hi @JavaZeroo, could you please share one data then I can take a look at the error?

**Reply by JavaZeroo** [reply]
[00.tif.zip](https://github.com/Project-MONAI/MONAI/files/11349681/00.tif.zip)

**Reply by JavaZeroo** [reply]
Which is from [Kaggle](https://www.kaggle.com/competitions/vesuvius-challenge-ink-detection/data)
